Sidekiq will now retry and add backtraces

williampratt opened the pull request:
Sidekiq will now retry and add backtraces even if the job is enqueued via another adapter.

Note: it does honor the retry and backtrace settings if already set

Commit:
Sidekiq will now retry and add backtraces even if the job is enqueued via another adapter.
Note: it does honor the retry and backtrace settings if already set

fixed tabs to spaces

ugh, editor

bleonard commented on lib/sidekiq_bus/middleware/retry.rb:
```diff
@@ -0,0 +1,29 @@
+      # ensure sidekiq will retry jobs even when they are enqueued via other adapters
+      class Retry
+        def call(worker, msg, queue)
+          msg['retry'] = true unless msg['retry']
```
shouldn't we do this only for our worker class?

williampratt replied:
No, I can explain tomorrow. It needs to be done in both.

williampratt replied:
Don't merge this yet. I'm going to try something now that I made a change in the initializer in metrics.

bleonard commented on lib/sidekiq_bus/middleware/retry.rb:
```diff
@@ -0,0 +1,29 @@
+      # ensure sidekiq will retry jobs even when they are enqueued via other adapters
+      class Retry
+       def call(worker_class, job, queue, redis_pool)
+         job['retry'] = Customer.current_id = true unless job['retry']
```
What's this `Customer` class for?

williampratt replied:
lol, I'll fix that. Must have been a copy / paste issue.

bleonard commented on lib/sidekiq_bus/middleware/retry.rb:
```diff
@@ -0,0 +1,29 @@
+      class Retry
+        def call(worker, msg, queue)
+          msg['retry'] = true unless msg['retry']
+          msg['backtrace'] = true unless msg['backtrace']
```
This is just always setting it to `true` - did you mean do `unless msg.has_key?('backtrace')` and similiar on others?

williampratt replied:
yep, will fix

bleonard commented on lib/sidekiq_bus/version.rb:
```diff
@@ -1,3 +1,3 @@
 module SidekiqBus
-  VERSION = "0.5.7"
+  VERSION = "0.5.8"
```
maybe do an `0.5.8.rc` next.

williampratt replied:
ok

Commit:
remove copy / paste error and add proper check for hash key

missed a (

Commit:
remove server side middleware that is not needed
add a filter in the client middleware to only act on events with class of QueueBus::Worker

bleonard commented:
looks good. thanks!

bleonard commented on lib/sidekiq_bus/version.rb:
```diff
@@ -1,3 +1,3 @@
 module SidekiqBus
-  VERSION = "0.5.7"
+  VERSION = "0.5.8rc"
```
usually `.rc`

Commit:
missed a '.' in version

gfoster commented:
Hey @bleonard what is outstanding on this that is preventing a merge?
